core/src/main/java/com/datastax/oss/driver/internal/core/cql/DefaultBoundStatement.java

```diff
@@ -358,7 +358,7 @@ public ByteBuffer getRoutingKey() {
       if (indices.isEmpty()) {
         return null;
       } else if (indices.size() == 1) {
-        return getBytesUnsafe(indices.get(0));
+        return isSet(0) ? getBytesUnsafe(indices.get(0)) : null;
```

Comment:
I don't think you can use "isSet(0)" here.  The first index could be somewhere else depending on the query... pretty sure this needs to be:

```java
      } else if (indices.size() == 1) {
        int index = indices.get(0);
        return isSet(index) ? getBytesUnsafe(index) : null;
      } else {
```

Reply:
ah right, sorry I missed that, thanks